payments-ui-core/src/test/java/com/stripe/android/ui/core/elements/CardNumberControllerTest.kt

```diff
@@ -466,16 +466,21 @@ internal class CardNumberControllerTest {
             }
         }
 
+        // Start by typing in an Amex card then a Mastercard
         cardNumberController.onValueChange("37")
-        verify(disallowedEventReporter, times(1)).onDisallowedCardBrandEntered(CardBrand.AmericanExpress)
-
         cardNumberController.onValueChange("372")
-        // Don't log again that we have disallowed American Express again
-        verify(disallowedEventReporter, times(1)).onDisallowedCardBrandEntered(CardBrand.AmericanExpress)
-
         cardNumberController.onValueChange("")
         cardNumberController.onValueChange("5555")
-        verify(disallowedEventReporter, times(1)).onDisallowedCardBrandEntered(CardBrand.MasterCard)
+
+        // Verify that AmericanExpress was reported once
+        val americanExpressReports = fakeDisallowedEventReporter.reportedBrands
+            .filter { it == CardBrand.AmericanExpress }
```

Comment:
This makes me question exactly what this test is doing, I still think it would be a better test with a turbine.

Reply:
Updated this to use a Turbine! Just as context the test was originally based off the number completed test: https://github.com/stripe/stripe-android/blob/master/payments-ui-core/src/test/java/com/stripe/android/ui/core/elements/CardNumberControllerTest.kt#L401

Reply:
Makes sense :/